src/proteus/dswx_hls.py

```diff
@@ -1116,38 +1196,35 @@ def _compute_diagnostic_tests(blue, green, red, nir, swir1, swir2,
     diagnostic_layer = np.zeros(shape, dtype = np.uint16)
 
     logger.info('step 1 - compute diagnostic tests')
-    for i in range(shape[0]):
-        for j in range(shape[1]):
-
-            # Surface water tests (see [1, 2])
-
-            # Test 1 (open water test, more conservative)
-            if (mndwi[i, j] > hls_thresholds.wigt):
-                diagnostic_layer[i, j] += 1
-
-            # Test 2 (open water test)
-            if (mbsrv[i, j] > mbsrn[i, j]):
-                diagnostic_layer[i, j] += 2
-
-            # Test 3 (open water test)
-            if (awesh[i, j] > hls_thresholds.awgt):
-                diagnostic_layer[i, j] += 4
-
-            # Test 4 (partial surface water test)
-            if (mndwi[i, j] > hls_thresholds.pswt_1_mndwi and
-                    swir1[i, j] < hls_thresholds.pswt_1_swir1 and
-                    nir[i, j] < hls_thresholds.pswt_1_nir and
-                    ndvi[i, j] < hls_thresholds.pswt_1_ndvi):
-                diagnostic_layer[i, j] += 8
-
-            # Test 5 (partial surface water test)
-            if (mndwi[i, j] > hls_thresholds.pswt_2_mndwi and
-                    blue[i, j] < hls_thresholds.pswt_2_blue and
-                    swir1[i, j] < hls_thresholds.pswt_2_swir1 and
-                    swir2[i, j] < hls_thresholds.pswt_2_swir2 and
-                    nir[i, j] < hls_thresholds.pswt_2_nir):
-                diagnostic_layer[i, j] += 16
-
+    # Surface water tests (see [1, 2])
+
+    # Test 1 (open water test, more conservative)
```

Comment:
It might reduce readability, but would improve performance to skip the np.where(..) step and simply put the existing conditionals inside the arrays' brackets. (see the comment on line 138 for an example.)